assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,87 @@
-console.log('***** Music Collection *****')
+console.log('***** Music Collection *****');
+
+// global variabls
+let collection = [];
+
+// functions
+let addToCollection = (title, artist, yearPublished) => {
+    let album = {
+        title: title,
+        artist: artist,
+        yearPublished: yearPublished,
+    };
+    collection.push(album);
+    return album;
+};
+
+let showCollection = (albumArray) => {
+    console.log(`Number of albums: ${albumArray.length}`);
+    for (let album of albumArray) {
+        console.log(
+            `${album.title} by ${album.artist},` +
+                ` released in ${album.yearPublished}`
+        );
+    }
+};
+
+let findByArtist = (artist) => {
+    let artistCollection = [];
+    for (let album of collection) {
+        if (album.artist.toLowerCase() === artist.toLowerCase()) {
+            artistCollection.push(album);
+        }
+    }
+    return artistCollection;
+};
+
+let search = (criteria) => {
+    let matchingAlbums = [];
+    for (let album of collection) {
+        let isMatching = Object.entries(criteria).reduce(
+            (m, [key, value]) =>
+                Object.hasOwn(album, key) && album[key] === value && m,
```

Comment:
I like this implementation. Allowing folks to search with any partial object is very cool and could get spun out into a general utility that searches _any_ array of objects.

That said, read those requirements carefully! The requirements specify an `{ artist, year }` object as a search criteria. Your collection objects have a `yearPublished` property, which means this (very cool) implementation technically does not match the requirements!